packages/mui-base/src/FocusTrap/FocusTrap.tsx

```diff
@@ -243,10 +268,7 @@ function FocusTrap(props: FocusTrapProps): JSX.Element {
       }
 
       // if the focus event is not coming from inside the children's react tree, reset the refs
-      if (
-        (nativeEvent && reactFocusEventTarget.current !== nativeEvent.target) ||
-        doc.activeElement !== reactFocusEventTarget.current
-      ) {
+      if (doc.activeElement !== reactFocusEventTarget.current) {
```

Comment:
The second part of the condition should always capture the first part too. I can't find scenario where the focusin's `event.target` will not be the `document.activeElement`

Reply:
I don't see a need for it either. As far as I now, `doc.activeElement` moves synchronously, so `nativeEvent.target === document.activeElement` at all time. Can't see a need for it from #21610.